lib/linter/node-event-generator.js

```diff
@@ -281,35 +107,42 @@ class NodeEventGenerator {
 	constructor(emitter, esqueryOptions) {
 		this.emitter = emitter;
 		this.esqueryOptions = esqueryOptions;
-		this.currentAncestry = [];
-		this.enterSelectorsByNodeType = new Map();
-		this.exitSelectorsByNodeType = new Map();
-		this.anyTypeEnterSelectors = [];
-		this.anyTypeExitSelectors = [];
 
 		emitter.eventNames().forEach(rawSelector => {
-			const selector = parseSelector(rawSelector);
+			const selector = parse(rawSelector);
 
-			if (selector.listenerTypes) {
+			/*
+			 * If this selector has identified specific node types,
+			 * add it to the map for these node types for faster lookup.
+			 */
+			if (selector.nodeTypes) {
 				const typeMap = selector.isExit
 					? this.exitSelectorsByNodeType
 					: this.enterSelectorsByNodeType;
 
-				selector.listenerTypes.forEach(nodeType => {
+				selector.nodeTypes.forEach(nodeType => {
 					if (!typeMap.has(nodeType)) {
 						typeMap.set(nodeType, []);
 					}
 					typeMap.get(nodeType).push(selector);
 				});
 				return;
 			}
+
+			/*
+			 * All selectors are added to the "any type" selectors
```

Comment:
```suggestion
			 * Remaining selectors are added to the "any type" selectors
```

Those added in `if (selector.nodeTypes) {` above are not added here.